src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -28,8 +28,12 @@ private ProjectId(string debugName)
             _debugName = debugName;
         }
 
-        internal ProjectId(Guid guid, string debugName)
+        public ProjectId(Guid guid, string debugName = null)
```

Comment:
Oh I see, this one lets you specify the GUID. Rather than making this public, add another overload of CreateNewId.